lib/identity_cache/query_api.rb

```diff
@@ -305,6 +307,7 @@ def find_batch(ids)
         @id_column ||= columns.detect {|c| c.name == primary_key}
         ids = ids.map{ |id| connection.type_cast(id, @id_column) }
         records = where(primary_key => ids).includes(cache_fetch_includes).to_a
+        records.map(&:readonly!) if IdentityCache.fetch_read_only_records && IdentityCache.should_use_cache?
```

Comment:
The result if the map isn't used, so I would use each instead: `records.each(&:readonly!)`